tests/test_frontend_op_resolver.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import xdsl.frontend.dialects.builtin as builtin
+
+from xdsl.dialects.arith import Addi, Constant, Mulf
+from xdsl.dialects.builtin import Float32Type, Float64Type, IntegerType, i32, i64, f32, f64
+from xdsl.frontend.exception import FrontendProgramException
+from xdsl.frontend.op_resolver import OpResolver
+
+
+def test_raises_exception_on_unknown_op():
+    with pytest.raises(FrontendProgramException) as err:
+        op = OpResolver.resolve_op("xdsl.frontend.dialects.arith", "unknown")
+    assert err.value.msg == "Internal failure: operation 'unknown' does not exist in module 'xdsl.frontend.dialects.arith'."
+
+
+def test_raises_exception_on_unknown_overload():
+    with pytest.raises(FrontendProgramException) as err:
+        op = OpResolver.resolve_op_overload("__unknown__", builtin._Integer)
+    assert err.value.msg == "Internal failure: '_Integer' does not overload '__unknown__'."
+
+
+def test_resolves_ops():
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(2, i32)
+
+    addi = OpResolver.resolve_op("xdsl.frontend.dialects.arith", "addi")
+    addi_op = addi(a, b)
+
+    assert isinstance(addi_op, Addi)
+    assert addi_op.operands[0] == a.results[0]
+    assert addi_op.operands[1] == b.results[0]
+    assert isinstance(addi_op.results[0].typ, IntegerType)
+    assert addi_op.results[0].typ.width.data == 32
+
+    c = Constant.from_float_and_width(5.0, f32)
+    mulf = OpResolver.resolve_op("xdsl.frontend.dialects.arith", "mulf")
+    mulf_op = mulf(c, c)
+
+    assert isinstance(mulf_op, Mulf)
+    assert mulf_op.operands[0] == c.results[0]
+    assert mulf_op.operands[1] == c.results[0]
+    assert isinstance(mulf_op.results[0].typ, Float32Type)
+
+
+def test_resolves_overloads():
+    a = Constant.from_int_and_width(1, i64)
+    b = Constant.from_int_and_width(2, i64)
+
+    addi = OpResolver.resolve_op_overload("__add__", builtin._Integer)
+    addi_op = addi(a, b)
+
+    assert isinstance(addi_op, Addi)
+    assert addi_op.operands[0] == a.results[0]
+    assert addi_op.operands[1] == b.results[0]
+    assert isinstance(addi_op.results[0].typ, IntegerType)
+    assert addi_op.results[0].typ.width.data == 64
+
+    c = Constant.from_float_and_width(5.0, f64)
+    mulf = OpResolver.resolve_op_overload("__mul__", builtin._Float64)
+    mulf_op = mulf(c, c)
+
+    assert isinstance(mulf_op, Mulf)
+    assert mulf_op.operands[0] == c.results[0]
+    assert mulf_op.operands[1] == c.results[0]
+    assert isinstance(mulf_op.results[0].typ, Float64Type)
```

Comment:
> Very Nice! I'm wondering if using decorators could make things easier compare to naming the function `resolve_X`? That way, we would make this resolution more explicit. What would you think about it? Otherwise I'm fine merging it!

I think medium to long term, the decorator approach sounds much nicer, but I think for now we can merge it as is and revisit this again in the future

Reply:
Yep, that's fine for me to merge as it is!

Reply:
(Just add the newline though ;) )